### ISSUE TITLE
Nx should not deduplicate  root `package.json` directly

### ISSUE THREAD

**Issue by houfeng0923** [nrwl/nx#13015]
## Current Behavior
When the root package: @xx/xxx  is a part of repo in mono-repo (it means the root src is a lib too, not just only in `packages` folder),
So, this monorepo now work well with lerna version, and publish . 
But, when add `lerna add-caching`, and run `npx lerna run test`, it can't find the root package:
```
Error: Invalid projects: @xx/xxx
```
Because It filtered the project file:`package.json` [code](https://github.com/nrwl/nx/blob/master/packages/nx/src/config/workspaces.ts#L738) directly

## Expected Behavior

when `pnpm-workspace.yaml` has config below:
```
packages:
  - '.'
  - 'packages/*'
  - '!dist/**'
```
Nx need reserve the root package.json in nx workspace.

## Steps to Reproduce

``` 
// has a mon-repo with pnpm-workspace.yaml 
packages:
  - '.' 
  - 'packages/*'
  - '!dist/**'
```
 
### Environment

```

   Node : 16.13.0
   OS   : darwin x64
   pnpm : 7.14.0
   
   nx : 15.0.10
   @nrwl/angular : Not Found
   @nrwl/cypress : Not Found
   @nrwl/detox : Not Found
   @nrwl/devkit : 15.0.8
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : Not Found
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : Not Found
   @nrwl/js : Not Found
   @nrwl/linter : Not Found
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : 15.0.2
   @nrwl/nx-plugin : Not Found
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/rollup : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : Not Found
   @nrwl/web : Not Found
   @nrwl/webpack : Not Found
   @nrwl/workspace : Not Found
   typescript : 4.8.2
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:
         lerna: 6.0.2

```

**Comment by AgentEnder** [comment]
Nx doesn't handle nested projects correctly currently, but we are working on it! This should be something fixed in the next 2 months or so.

**Comment by AgentEnder** [comment]
This should be fixed by #13074
